src/bplus_tree.rs

```diff
@@ -0,0 +1,525 @@
+                        node.children.push(None);
+                        let mut temp1 = node.children.swap_remove(i + 1).unwrap();
+                        self.merge(&mut temp, &mut temp1);
+                        node.key_num -= 1;
+                        let key_num = node.key_num;
+                        for j in 0..key_num {
+                            node.keys[j] = node.keys[j + 1].clone();
+                            node.keys[j] = node.keys[j + 1].clone();
+                        }
+                        node.keys.resize(key_num, K::default());
+                        node.children[i + 1] = Some(temp1);
+                    } else {
+                        node.children.push(None);
+                        let mut prev_children = node.children.swap_remove(i + 1).unwrap();
+                        let moved_key = prev_children.keys[0].clone();
+                        let moved_value = prev_children.pointers[0].clone();
+                        self.insert_helper(&mut temp, &moved_key, moved_value);
+                        self.remove_helper(&mut prev_children, &moved_key);
+                        node.keys[i] = prev_children.keys[0].clone();
+                        node.children[i + 1] = Some(prev_children);
+                    }
+                }
+
+                if temp.key_num == 0 {
+                    node.children[i] = temp.children.swap_remove(0);
+                } else {
+                    node.children[i] = Some(temp);
+                }
+            }
+        }
+    }
+
+    fn merge(&self, node1: &mut Box<Node<K>>, node2: &mut Box<Node<K>>) {
+        let new_key_num = node1.key_num + node2.key_num;
+
+        node1.keys.resize(new_key_num, K::default());
+        if node1.leaf {
+            node1.pointers.resize(new_key_num, ChunkHandler::default());
+        } else {
+            node1.children.resize(new_key_num + 2, None);
+        }
+        for i in 0..node2.key_num {
+            let temp = node1.key_num;
+            node1.keys[temp + i] = node2.keys[i].clone();
+            if node1.leaf {
+                node1.pointers[temp] = node2.pointers[i].clone();
+            } else {
+                node1.children[temp + i + 1] = node2.children[i].clone();
+            }
+        }
+
+        if !node1.leaf {
+            node1.children[new_key_num + 1] = node2.children[node2.key_num].clone();
+        }
+    }
+}
+
+impl<K: Ord + Clone + Default> Database<K, DataContainer<()>> for BPlus<K> {
+    fn insert(&mut self, key: K, value: DataContainer<()>) -> io::Result<()> {
+        match value.extract() {
+            Data::Chunk(chunk) => self.insert(key, chunk.clone()),
+            Data::TargetChunk(_chunk) => unimplemented!(),
+        }
+    }
+
+    fn get(&self, key: &K) -> io::Result<DataContainer<()>> {
+        match self.get(key) {
+            Ok(chunk) => Ok(DataContainer::from(chunk)),
+            Err(error) => Err(error),
+        }
```

Comment:
```suggestion
        self.get(key).map(DataContainer::from)
```
вот такие красивые штуки есть у Result, у Option есть похожие, а у итераторов несчетное количество методов, из которых самый важный это тоже map